sound/soc/sof/intel/Kconfig

```diff
@@ -276,6 +276,7 @@ config SND_SOC_SOF_HDA_LINK
 	bool "SOF support for HDA Links(HDA/HDMI)"
 	depends on SND_SOC_SOF_NOCODEC=n
```

Comment:
@ranj063 everything looks good! I wonder if we can have a common driver for DMIC/SSP/ESAI/SAI. After all the code so far looks pretty much the same. The only difference is the way we handle the machine driver.

Reply:
@dbaluta do you mean the same platform device for all these types? 
DMIC and SSP device probes are different in the sense that the SSP probe checks for acpi match where the DMIC probe checks for DMIC presence in the NHLT table.

Reply:
No, I mean sof-ssp-audio.c, sof-dmic-audio.c and sof-hda-audio.c could be made one single generic file and the probe function will be called multiple times for each instance of DMIC / SSP / HDA.
I see now that DMIC is pretty different.

I was just looking only at sof-hda-audio.c and sof-ssp-audio.c and they seem pretty similar. 

But I guess we can go with this implementation now and maybe refactor it later.

Reply:
@ranj063 one question. Can SSP audio client can also support recording? Or that is just done with DMIC only?

Reply:
@dbaluta yes all the ssp pipelines will be define in the same topology.